packages/compose/src/hooks/use-copy-on-click/index.js

```diff
@@ -30,45 +25,67 @@ export default function useCopyOnClick( ref, text, timeout = 4000 ) {
 		alternative: 'wp.compose.useCopyToClipboard',
 	} );
 
-	/** @type {import('react').MutableRefObject<Clipboard | undefined>} */
-	const clipboard = useRef();
 	const [ hasCopied, setHasCopied ] = useState( false );
 
 	useEffect( () => {
 		/** @type {number | undefined} */
 		let timeoutId;
+		/** @type Array<Element>) */
+		let triggers = [];
 
 		if ( ! ref.current ) {
 			return;
 		}
 
-		// Clipboard listens to click events.
-		clipboard.current = new Clipboard( ref.current, {
-			text: () => ( typeof text === 'function' ? text() : text ),
-		} );
+		// `triggers` is always an array, regardless of the value of the `ref` param.
+		if ( typeof ref.current === 'string' ) {
+			// expect `ref` to be a DOM selector
+			triggers = Array.from( document.querySelectorAll( ref.current ) );
+		} else if ( 'length' in ref.current ) {
+			// Expect `ref` to be a `NodeList`
+			triggers = Array.from( ref.current );
+		} else {
+			// Expect `ref` to be a single `Element`
+			triggers = [ ref.current ];
+		}
```

Comment:
Previously, the different values for `ref.current` were handles by `Clipboard`'s constructor